File: app/src/main/java/org/wikipedia/dataclient/okhttp/TestStubInterceptor.java
package org.wikipedia.dataclient.okhttp;

import java.io.IOException;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import okhttp3.Interceptor;
import okhttp3.Response;

public class TestStubInterceptor implements Interceptor {
    public interface Callback {
        Response getResponse(Chain request) throws IOException;
    }

    @Nullable private static Callback CALLBACK;

    public static void setCallback(@Nullable Callback callback) {
        CALLBACK = callback;
    }

    @Override
    public Response intercept(@NonNull Chain chain) throws IOException {
        if (CALLBACK != null) {
            return CALLBACK.getResponse(chain);
        }
        return chain.proceed(chain.request());
    }
}

File: app/src/main/java/org/wikipedia/dataclient/okhttp/TestStubInterceptor.kt
package org.wikipedia.dataclient.okhttp

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class TestStubInterceptor : Interceptor {
    interface Callback {
        @Throws(IOException::class)
        fun getResponse(request: Interceptor.Chain): Response
    }

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        return if (CALLBACK != null) {
            CALLBACK!!.getResponse(chain)
        } else chain.proceed(chain.request())
    }

    companion object {
        var CALLBACK: Callback? = null
    }
}
